### PR TITLE
Homework10: Anastasiya Solodukho

### PR DESCRIPTION
**Pull request by AnastasiyaSo**
Added HW10

### CODE REVIEW

File: homework_10.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# Task #1
+
+
+def validate_arguments(func):
+    """Function checks whether the arguments are positive numbers"""
+    def inner(*arg):
+        for i in arg:
+            if i > 0:
+                continue
+            raise ValueError
```
Comment (Stml89) [inline homework_10.py]:
```suggestion
            if i < 0:
                raise ValueError(<error message>)
            ...

```

File: homework_10.py
Comment (Stml89) [inline homework_10.py:1]: Please put each task into a separate file

File: homework_10.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+assert new_func(2, -5), ValueError
+
+# Task #2
+
+
+def is_integer(func):
+    """Function checks whether the sum of the arguments is a number"""
+    def inner(a, b):
+        c = a + b
+        func(a, b)
```
Comment (Stml89) [inline homework_10.py]:
```suggestion
        result = func(a, b)
```

File: homework_10.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    _str += str(i)
+                return _str
+            if _type == int:
+                for i in arg:
+                    _int += int(i)
+                return _int
+            if _type == float:
+                for i in arg:
+                    _float += float(i)
+                return _float
```
Comment (Stml89) [inline homework_10.py]:
```suggestion
    converted_args = map(_type, args)
    func(converted_args)
```

File: homework_10.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            return True
+        return inner
+    return decorator
+
+
+@typed(_type=str)
+def add_str(*arg):
+    """Function that passes an argument"""
+    return arg
```
Comment (Stml89) [inline homework_10.py]:
```suggestion
def add_str(a, b):
    """Function that passes an argument"""
    return a + b
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Comment by AnastasiyaSo** [comment]
@Stml89 fixed. Please, check

File: homework_10_2.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# Task #2
+
+
+def is_integer(func):
+    """Function checks whether the sum of the arguments is a number"""
+    def inner(a, b):
+        result = func(a, b)
+        if isinstance(result, (int, float)):
+            return True
+        return f"{result} is not a number"
```
Comment (Stml89) [inline homework_10_2.py]: Better to return a `bool` instead of `str`

File: homework_10_1.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def validate_arguments(func):
+    """Function checks whether the arguments are positive numbers"""
+    def inner(*arg):
+        for i in arg:
+            if i <= 0:
+                raise ValueError("Value is not positive")
+        func(*arg)
+        return True
```
Comment (Stml89) [inline homework_10_1.py]:
```suggestion
        return func(*arg)
```

File: homework_10_3.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    def decorator(func):
+        def inner(*arg):
+            func(*arg)
+            _str = ""
+            converted_args = list(map(_type, arg))
+            if _type == str:
+                for i in converted_args:
+                    _str += str(i)
+                return _str
+            sum_numbers = sum(converted_args)
```
Comment (Stml89) [inline homework_10_3.py]: This functionality should be done in `add_str`, `add_int` and etc

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Comment by AnastasiyaSo** [comment]
@Stml89 fixed again. Please, check

**Review by Stml89 — APPROVED** [review]

**Pull request merged**